spatialprofilingtoolbox/apiserver/app/main.py

```diff
@@ -176,20 +174,38 @@ async def request_phenotype_proximity_computation(
     retrieve = query().retrieve_signature_of_phenotype
     criteria1 = retrieve(phenotype1, study)
     criteria2 = retrieve(phenotype2, study)
-    with CountRequester() as requester:
+    with OnDemandRequester() as requester:
         metrics = requester.get_proximity_metrics(
             query().get_study_components(study).measurement,
             radius,
-            [
+            (
                 criteria1.positive_markers,
                 criteria1.negative_markers,
                 criteria2.positive_markers,
                 criteria2.negative_markers,
-            ],
+            ),
         )
     return metrics
 
 
+@app.get("/request-squidpy-computation/")
+async def request_squidpy_computation(
+    study: ValidStudy,
+    phenotypes: list[ValidPhenotype],
+) -> SquidpyMetricsComputationResult:
+    """Spatial proximity statistics between phenotype clusters as calculated by Squidpy."""
+    criteria: list[PhenotypeCriteria] = [
+        query().retrieve_signature_of_phenotype(p, study) for p in phenotypes]
```

Comment:
Again, I think I will systematically change the open-closed parens or braces pattern to the one that creates scope blocks that are easily visible without reading the block contents. In this case:
```py
    criteria: list[PhenotypeCriteria] = [
        query().retrieve_signature_of_phenotype(p, study) for p in phenotypes
    ]
```
The point is that, upon seeing/encountering the `[` character, there is a non-recursive algorithm for finding its matching brace `]` that does not require pushing this thought onto a stack and reading the entirety of the enclosed block. Namely, go to the first character of the line (in this case the `c` in criteria), then go down until `]` is encountered.
There is no such algorithm in the alternative form, besides "scan every character until `]` is encountered, unless there are other `[`, in which case increase a mental counter, then scan until a `]` is encountered, unless there are other `[`or `]`, in which case increment or decrement the mental counter accordingly, then scan until ...". This is unnecessary stress in our lives.

Reply:
(Obviously, I am disregarding likely conflicts with style mandates. If they do conflict, they probably do not take into account the consideration above, which is more important to me than global style compliance.)

Reply:
(This gives me the idea that source code is actually not usually stored in quite the right data structure; source code is really a matrix of characters, not a sequence.)